Realm/RLMManagedSet.mm

```diff
@@ -507,32 +507,27 @@ - (BOOL)isFrozen {
     return _realm.isFrozen;
 }
 
-- (instancetype)freeze {
-    if (self.frozen) {
-        return self;
-    }
-
-    RLMRealm *frozenRealm = [_realm freeze];
-    auto& parentInfo = _ownerInfo->resolve(frozenRealm);
+- (instancetype)resolveInRealm:(RLMRealm *)realm {
```

Comment:
It would be nice to at some stage create a common home for all of these repeated methods in the collections